src/bblocks_data_importers/who/ghed.py

```diff
@@ -0,0 +1,190 @@
+
+    @staticmethod
+    def _format_raw_data(raw_data: io.BytesIO) -> pd.DataFrame:
+        """Format the raw data from the GHED database
+
+        Args:
+            raw_data: Raw data extracted from the GHED database. Use the `_extract_raw_data` method to get this data
+        """
+
+        try:
+            data_df = (
+                pd.read_excel(raw_data, sheet_name="Data")
+                .drop(columns=["region", "income"])
+                .melt(id_vars=["country", "code", "year"], var_name="indicator_code")
+                .rename(columns={"country": "country_name", "code": "iso3_code"})
+            )
+
+            codes_df = (
+                pd.read_excel(raw_data, sheet_name="Codebook")
+                .rename(
+                    columns={
+                        "variable code": "indicator_code",
+                        "variable name": "indicator_name",
+                    }
+                )
+                .loc[:, ["indicator_code", "indicator_name", "unit", "currency"]]
+                .replace("-", np.nan)
+            )
+
+            return pd.merge(data_df, codes_df, on="indicator_code", how="left")
+
+        except (ValueError, KeyError) as e:
+            raise DataFormatError(f"Error formatting data: {e}")
+
+    @staticmethod
+    def _format_metadata(raw_data: io.BytesIO) -> pd.DataFrame:
+        """Format the metadata from the GHED database
+
+        Args:
+            raw_data: Raw data extracted from the GHED database. Use the `_extract_raw_data` method to get this data
+        """
+
+        cols = {
+            "country": "country_name",
+            "code": "iso3_code",
+            "variable name": "indicator_name",
+            "variable code": "indicator_code",
+            "Sources": "sources",
+            "Comments": "comments",
+            "Data type": "data_type",
+            "Methods of estimation": "methods_of_estimation",
+            "Country footnote": "country_footnote",
+        }
+
+        try:
+            return (
+                pd.read_excel(raw_data, sheet_name="Metadata")
+                .rename(columns=cols)
+                .loc[:, cols.values()]
+            )
+
+        except (ValueError, KeyError) as e:
+            raise DataFormatError(f"Error formatting metadata: {e}")
+
+    def _load_data(self):
+        """Load the data from the GHED database to the object"""
+
+        self._raw_data = self._extract_raw_data()
+        self._data = self._format_raw_data(self._raw_data)
+        self._metadata = self._format_metadata(self._raw_data)
+
+    def get_data(self) -> pd.DataFrame:
+        """Get the data from the GHED database
+
+        # TODO: Add functionality to filter the data by country, indicator, year, etc.
+
+        Returns:
+            A DataFrame with the data from the GHED database
+        """
+
+        if self._data is None:
+            self._load_data()
+
+        return self._data
```

Comment:
if it were a partially implemented ABC, we could add that functionality (in your TODO) there.

get_data would be pretty predictable (if _data is None, load it using the private pipeline method), and we could add additional things like what you're describing there directly to the base class.

Thoughts?